src/Protocol/ActivityPub/Receiver.php

```diff
@@ -487,6 +487,10 @@ public static function processActivity($activity, string $body = '', int $uid =
 					$object_data['thread-completion'] = true;
 
 					$item = ActivityPub\Processor::createItem($object_data);
+					if (empty($item)) {
+						return;
```

Comment:
Shouldn't we log something here?

Reply:
I'm unsure, since in `createItem` we already logged it:
`2020-09-19T17:02:38Z index [INFO]: Parent not found, message will be discarded. {"thr-parent":"http...`

Reply:
Ok. Ideally, `createItem` should throw an Exception and rely on the calling code to handle it because it shouldn't have to know about the context this method was called in.

Not something you should change right now, I'm just trying to distill better programming knowledge little by little until you feel like adopting it yourself.